src/views/Users.js

```diff
@@ -0,0 +1,464 @@
+		setUsers(userCopy);
+	};
+
+	const InnerPageNumbers = () => {
+		let default_btn =
+			"z-10 bg-white border-gray-300 text-gray-700 relative inline-flex items-center px-4 py-2 border  hover:bg-gray-50";
+		let clicked_btn =
+			"z-10 bg-gray-50 border-gray-500 text-gray-600 relative inline-flex items-center px-4 py-2 border  hover:bg-gray-50";
+		let menuItems = [];
+		if (pageCount < 6) {
+			for (let i = 0; i < pageCount; i++) {
+				menuItems.push(
+					<button
+						className={pageIndex === i ? clicked_btn : default_btn}
+						onClick={() => gotoPage(i)}
+					>
+						{i + 1}
+					</button>
+				);
+			}
+		} else {
+			let j =
+				pageIndex === 0 || pageIndex === 1
+					? 0
+					: pageCount - pageIndex === 1 ||
+					  pageCount - pageIndex === 0 ||
+					  pageCount - pageIndex === 2
+					? pageCount - 5
+					: pageIndex - 2;
+			for (let i = j; i < j + 5; i++) {
+				menuItems.push(
+					<button
+						className={pageIndex === i ? clicked_btn : default_btn}
+						onClick={() => gotoPage(i)}
+					>
+						{i + 1}
+					</button>
+				);
+			}
+		}
+		return <div>{menuItems}</div>;
+	};
+
+	const refresh = (filter) => {
+		getUsers(getAccessTokenSilently)
+			.then((list) => {
+				setEmpty(list.length === 0);
+				return list;
+			})
+			.then((sorted) => {
+				return filter != null ? filter(sorted) : sorted;
+			})
+			.then((filtered) => {
+				setUsersBackup(filtered);
+				setUsers(filtered);
+			})
+			.then(() => setLoading(false))
+			.catch((error) => handleError(error));
+	};
+
+	useEffect(() => {
+		refresh(null);
+	}, [getAccessTokenSilently]);
+	useEffect(() => {}, [users]);
+
+	return (
+		<div className="py-10 bg-gray-50 overflow-hidden rounded max-w-7xl mx-auto sm:px-6 lg:px-8">
+			<header className="sm:flex sm:items-center">
+				<div className="max-w-7xl mx-auto sm:px-6 lg:px-8 sm:flex-auto">
+					<h1 className="text-xl font-semibold text-gray-900">
+						Users
+					</h1>
+					<p className="mt-2 text-sm text-gray-700">
+						A list of all Users
+					</p>
+				</div>
+			</header>
+			<main>
+				<div className="max-w-7xl mx-auto">
+					<div className="px-4 sm:px-0">
+						{loading && <Loading />}
+						{error != null && <span>{error.toString()}</span>}
+						<main>
+							{loading && <Loading />}
+							{error != null && <span>{error.toString()}</span>}
+
+							{!empty ? (
+								<div className="max-w-7xl mx-auto sm:px-6 lg:px-8">
+									<div className="flex w-full items-center mt-8 justify-between">
+										<div className="flex">
+											<input
+												className="text-sm rounded p-2 border border-gray-300 focus:border-gray-400 outline-none w-[300px]"
+												placeholder="Search..."
+												type="search"
+												onChange={(e) =>
+													handleSearch(e.target.value)
+												}
+											/>
+											<div className="flex items-center mx-auto sm:px-6 lg:px-8">
+												<p className="ml-6 text-sm text-gray-700 px-4">
+													Sort by: &nbsp;
```

Comment:
the values there are the ones we need or it was a copy from peers? What can we sort by?